src/com/sahokia/kos/bustickets/validation/BusTicketsValidator.java

```diff
@@ -0,0 +1,134 @@
+package com.sahokia.kos.bustickets.validation;
+
+import com.sahokia.kos.bustickets.exception.PriceException;
+import com.sahokia.kos.bustickets.exception.StartDateException;
+import com.sahokia.kos.bustickets.exception.TicketClassException;
+import com.sahokia.kos.bustickets.exception.TicketTypeException;
+import com.sahokia.kos.bustickets.model.BusTicket;
+import com.sahokia.kos.bustickets.model.TicketClass;
+import com.sahokia.kos.bustickets.model.TicketType;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BusTicketsValidator {
+    private static final String TICKET_CLASS_VIOLATION_NAME = "ticket class";
+    private static final String TICKET_TYPE_VIOLATION_NAME = "ticket type";
+    private static final String PRICE_VIOLATION_NAME = "price";
+    private static final String START_DATE_VIOLATION_NAME = "start date";
+
+    public void validateBusTickets(List<BusTicket> busTickets) {
+        Map<String, Boolean> currentBusTicketViolations;
+        int valid = 0;
+        Map<String, Integer> violations = new HashMap<>();
+        violations.put(TICKET_CLASS_VIOLATION_NAME, 0);
+        violations.put(TICKET_TYPE_VIOLATION_NAME, 0);
+        violations.put(PRICE_VIOLATION_NAME, 0);
+        violations.put(START_DATE_VIOLATION_NAME, 0);
+        for (BusTicket busTicket : busTickets) {
+            currentBusTicketViolations = validateBusTicket(busTicket);
+            if (!currentBusTicketViolations.containsValue(true)) {
+                valid++;
+            } else {
+                currentBusTicketViolations.forEach((key, value) -> {
+                    if (value) {
+                        violations.put(key, violations.get(key) + 1);
+                    }
+                });
+            }
+        }
+
+        String mostPopularViolation = violations.entrySet()
+                .stream()
+                .max(Map.Entry.comparingByValue())
+                .map(Map.Entry::getKey)
+                .orElse(null);
```

Comment:
Need to be careful with nulls due to further potential NPE, probably better to provide default value
It's up to you fix it or leave as is :)

Reply:
Done